rpc2/dispatch.go

```diff
@@ -6,9 +6,13 @@ type DecodeNext func(interface{}) error
 type ServeHook func(DecodeNext) (interface{}, error)
 
 type Dispatcher interface {
-	Dispatch(m *Message) error
 	Call(name string, arg interface{}, res interface{}, f UnwrapErrorFunc) error
 	RegisterProtocol(Protocol) error
+}
+
+type dispatcher interface {
+	Dispatcher
```

Comment:
I think this proposal misses a key point that I really want to preserve --- that `Client` and `Server` are both bound to a particular protocol. I really want to keep this.